src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
             return node.CreateBindingSuccess(value);
         }
 
-        return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        var bindingResult = MatchExpressionValue(node, expression);
+        if ( bindingResult is not null)
+        {
+            return bindingResult;
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        }
+
+        
     }
+
+    private HttpBindingResult<object?>? MatchExpressionValue(HttpExpressionNode node, string expression)
+    {
+
+        var guidPattern = new Regex(@$"\{"$guid"}", RegexOptions.Compiled);
+        var dateTimePattern = new Regex(@$"\{"$datetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
```

Comment:
qq: `$guid` and `$GUID` would mean different things, would they not? In other words, variables are always case-sensitive, aren't they?

Reply:
Correct, they're distinct but we're only allowing lowercase guid. Are you saying that we should support both?

Reply:
Nope was just checking.

Iirc, http primitives such as header names, urls etc. are not case sensitive - but it seems reasonable for variables in .http files to be case sensitive. I was just checking to confirm that this matches the spec (and the other http editors that we have looked at).